src/bin/bench-tps.rs

```diff
@@ -469,9 +468,23 @@ fn main() {
         time_sec = s.to_string().parse().expect("integer");
     }
 
-    if let Some(s) = matches.value_of("addr") {
-        addr = Some(s.to_string());
-    }
+    let addr = if let Some(s) = matches.value_of("addr") {
+        match s.to_string().parse() {
```

Comment:
nit: Whenever I see the `Ok(foo) => foo, Err(e) => ...` pattern I think of https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or_else myself, as a `match` feels a little to heavy.